feeds/wlan-ap/opensync/src/platform/openwrt/src/rrm/src/rrm_channel.c

```diff
@@ -224,9 +227,20 @@ void rrm_nf_timer_handler(struct ev_loop *loop, ev_timer *timer, int revents)
 
 			get_channel_bandwidth(get_max_channel_bw_channel(ieee80211_channel_to_frequency(rrm_config->rrm_data.backup_channel),
 					radio->schema.ht_mode), &channel_bandwidth);
+			/*
+			 * Update the Wifi_Radio_Config table with the "switch-to" channel.
+			 * This is done to avoid the mismatch between radio state and radio config
+			 * because a mismatch would cause the radio to fallback to the original channel.
+			 */
+			iter = ovsdb_where_simple(SCHEMA_COLUMN(Wifi_Radio_Config, if_name), radio->schema.if_name);
+			if (iter) {
+				if (ovsdb_table_select_one_where(&table_Wifi_Radio_Config, iter, &config))
+					config.channel = rrm_config->rrm_data.backup_channel;
+			}
```

Comment:
do we still need to do ubus call for channel switch, since if you change the radio_config it would change the channel.